### PR TITLE
feat: add yield-yak farms

### PR DESCRIPTION
**Pull request by 0xdapper**
## Checklist

- [x] I have followed the [Contributing Guidelines](../CONTRIBUTING.md)
- [x] (optional) As a contributor, my Ethereum address/ENS is: 0xdapper.eth
- [x] (optional) As a contributor, my Twitter handle is: 0xdapper_

## How to test?

http://localhost:5001/apps/yieldyak/balances?addresses[]=0xdb68270db840830d76ced4ab763fb045f1113b68&network=avalanche

### CODE REVIEW

**Comment by JForsaken** [comment]
Unfortunately, the changes you're trying to add aren't compatible to the new Studio patterns we're trying to push forward 😢  Since Yield Yak is migrated to the new implementation (template pattern), the farms you're adding also need to be using the template patterns (we do not support a mix of legacy & template on a single app, it's either all or nothing)

**Comment by JForsaken** [comment]
@0xdapper I'll take over your branch to do the migration directly, so you know exactly what's going on for the next time around!

**Comment by 0xdapper** [comment]
@JForsaken which part of my implementation is legacy and which is the new pattern?

**Comment by JForsaken** [comment]
@0xdapper I'll DM on discord gimme a min! easier lol

File: src/apps/yield-yak/helpers/yield-yak.vault.token-definitions-resolver.ts [hunk 1]
```diff
@@ -28,17 +32,62 @@ const query = gql`
...
-    return data.farms;
+  async getVaultDefinitionsData(network: Network): Promise<FarmsResponse['farms']> {
+    const farms = await Axios.get<YieldYakFarmDetails[]>('https://staging-api.yieldyak.com/farms').then(x => x.data);
+    const multicall = this.appToolkit.getMulticall(network);
+
+    const response = await Promise.all(
+      farms.map(async farm => {
+        const farmContract = this.yieldyakContractFactory.yieldYakVault({
+          address: farm.address,
+          network: Network.AVALANCHE_MAINNET,
```
Comment (JForsaken) [inline src/apps/yield-yak/helpers/yield-yak.vault.token-definitions-resolver.ts:66]: can reuse `network` from the fn params here

File: src/apps/yield-yak/helpers/yield-yak.vault.token-definitions-resolver.ts [hunk 2]
```diff
@@ -28,17 +32,62 @@ const query = gql`
...
+        });
+
+        try {
+          const [depositTokenBalance, name, depositToken] = await Promise.all([
```
Comment (JForsaken) [inline src/apps/yield-yak/helpers/yield-yak.vault.token-definitions-resolver.ts:70]: Just went over the code in the vault token fetcher -- I suggest you removing all those calls to multicall (to get `depositTokenBalance`, `name`, and `depositToken`), as we only need the `depositToken` to be used in the get underlying token step.

It would be greatly beneficial to use `multicall` in the `getUnderlyingTokenAddresses` directly, using `multicall` passed in the fn params. Doing it this way helps our system be efficient with all the on-chain calls of the whole fetcher.

So to summarize, you can simply return an array of all the farm addresses here in the definitions resolver, and in the fetcher, do the on-chain call to get the `depositToken` in the `getUnderlyingTokenAddresses` using `multicall` from its context 😄

File: src/apps/yield-yak/helpers/yield-yak.vault.token-definitions-resolver.ts (hunk 1 above)
Comment (JForsaken) [reply src/apps/yield-yak/helpers/yield-yak.vault.token-definitions-resolver.ts:66]: nevermind you can remove the contract altogether as referred in the other comment!

**Review by JForsaken — APPROVED** [review]
